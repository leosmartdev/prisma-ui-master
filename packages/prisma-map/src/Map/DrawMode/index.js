import DrawMode from './DrawMode';
import {
  convertModeToTool,
  convertToolToMode,
} from './tools';

export default DrawMode;

export {
  convertModeToTool,
  convertToolToMode,
};
