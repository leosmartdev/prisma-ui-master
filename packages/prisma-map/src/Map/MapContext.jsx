import React from 'react';

// Create the Map context to allow children to access the map object and call methods on map
export const { Provider, Consumer } = React.createContext(null);
