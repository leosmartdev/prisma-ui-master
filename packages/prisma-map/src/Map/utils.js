/*
 * COPYRIGHT Â© 2018 OROLIA GROUP AND/OR ITS AFFILIATES. ALL RIGHTS ARE STRICTLY RESERVED.
 * THIS SOURCE CODE CONTAINS PROPRIETARY AND CONFIDENTIAL INFORMATION AND DATA AND IS THE
 * SOLE PROPERTY OF OROLIA GROUP AND/OR ITS AFFILIATES.
 * THIS SOURCE CODE MUST NOT BE USED, DISSEMINATED, OR DISTRIBUTED EXCEPT FOR THE AGREED PURPOSE.
 * UNAUTHORIZED USE, REPRODUCTION, OR ISSUE TO ANY THIRD PARTY IS NOT PERMITTED WITHOUT THE
 * PRIOR WRITTEN CONSENT OF THE OROLIA GROUP.
 * THIS SOURCE CODE IS TO BE RETURNED TO THE OROLIA GROUP WHEN THE AGREED PURPOSE IS FULFILLED.
 * -------------------------------------
 * Utility functions for the map
 */
/* eslint-disable import/prefer-default-export */

/**
 * Gets the first feature in a list of features, or returns null if there are no features.
 *
 * @param {array(GeoJSON Features)} features
 * @return {object} The first feature, or null if there are no features.
 */
export function getFirstFeature(features) {
  if (features && features.length > 0) {
    const [feature] = features;
    return feature;
  }

  return null;
}
