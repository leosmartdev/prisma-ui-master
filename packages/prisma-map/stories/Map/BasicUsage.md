# Map Component

The Map component is a lightweight wrapper around the Uber `react-map-gl/ReactMapGL` component.
We provided a few new capabilites that fit within the Prisma use cases as well as providing
a layer of abstraction around mapbox so if we need to change engines in the future you only
have to change the import of the `Map` component not the interface itself.

We however, have decided on the Mapbox API for most of our interactions with the map. The use of
JSON is declarative and clean, as well as easily convertable to other mapping engines when needed.
This means, if you are used to the Mapbox way of doing things, our API will be almost identical.

There are a few minor changes and features that we will be adding to our API, and most of them
are related to moving features into Redux so we can dispatch map actions, as well as custom tools
like the measure and zones tools that utilize the map. 
