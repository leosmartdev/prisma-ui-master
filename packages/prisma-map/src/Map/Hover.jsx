/*
 * COPYRIGHT Â© 2018 OROLIA GROUP AND/OR ITS AFFILIATES. ALL RIGHTS ARE STRICTLY RESERVED.
 * THIS SOURCE CODE CONTAINS PROPRIETARY AND CONFIDENTIAL INFORMATION AND DATA AND IS THE
 * SOLE PROPERTY OF OROLIA GROUP AND/OR ITS AFFILIATES.
 * THIS SOURCE CODE MUST NOT BE USED, DISSEMINATED, OR DISTRIBUTED EXCEPT FOR THE AGREED PURPOSE.
 * UNAUTHORIZED USE, REPRODUCTION, OR ISSUE TO ANY THIRD PARTY IS NOT PERMITTED WITHOUT THE
 * PRIOR WRITTEN CONSENT OF THE OROLIA GROUP.
 * THIS SOURCE CODE IS TO BE RETURNED TO THE OROLIA GROUP WHEN THE AGREED PURPOSE IS FULFILLED.
 * -------------------------------------
 * Component for detecting hover events on specific map layers and rendering a child function
 * when the hover occurs.
 */
import React from 'react';
import PropTypes from 'prop-types';

import { Consumer as MapConsumer } from './MapContext';

export class MapHover extends React.Component {
  static propTypes = {
    /**
     * Map instance passed from the <Map.Consumer> context wrapper when using <Hover> or if using
     * `MapHover` directly, this must be an instance of `<Map>` component.
     *
     * If you use `<Hover>` or the reexported, `<Map.Hover>` from `@prisma/Map` then you can safely
     * ignore this property as it will be provided to `<MapHover>` automatically.
     */
    map: PropTypes.object.isRequired,

    /**
     * The children to render, must be a function that will be called whenever the hover occurs
     * over the provided layers. The hover event will happen when the mouseover event is triggered
     * by the mouse over a feature on the layers. The function will then be called whenever the map
     * is moved or the mouse is moved until mouseleave event occurs. At this point, the hover render
     * will return null instead of calling this child function to render nothing.
     *
     * ## Signature
     * ```
     * (features: array(object), event: object) => Component
     * ```
     *
     * @param {array} features Array of all features currently under the mouse cursor. Each feature
     *                         is a geojson object.
     * @param {object} event   The mouse event that started the hover, with additional mouse events
     *                         points added so the `event.point` property always contains the
     *                         latest x, y values of the mouse cursor and coordinates always has
     *                         latest lat, lng values of the mouse cursor as well. The features
     *                         array will hold the actual coordinates of the feature the house is
     *                         over where the event is the location of the mouse pointer itself.
     *
     *
     */
    children: PropTypes.func,

    /**
     * Array of all the layer ids this hover should listen for mouse events. When a layer id is
     * removed from this array, the component will stop listening to events on that layer, and when
     * a new layer id is added it will immeditately subscribe to events.
     * Unknown layer ids will be ignore and an empty array will unsubscribe from all events.
     */
    layers: PropTypes.arrayOf(PropTypes.string),
  };

  static defaultProps = {
    layers: [],
    children: null,
  };

  state = {
    isHovering: false,
    features: null,
    lastMouseEvent: null,
  };

  componentDidMount() {
    const { map } = this.props;
    if (!map) {
      return;
    }

    this.updateLayers();
  }

  componentDidUpdate(prevProps) {
    const { map, layers } = this.props;
    if (!map) {
      return;
    }

    // If the map changed we need to subscribe to all events on the new map
    if (prevProps.map !== map) {
      this.updateLayers();
    }

    if (prevProps.layers !== layers) {
      this.updateLayers(prevProps.layers);
    }
  }

  onMouseOver = event => {
    const { map } = this.props;

    map.on('move', this.onMove);

    this.setState({
      lastMouseEvent: event,
      isHovering: true,
      features: this.getFeatures(event),
    });
  };

  onMouseLeave = () => {
    const { map } = this.props;

    map.off('move', this.onMove);

    this.setState({
      features: null,
      lastMouseEvent: null,
      isHovering: false,
    });
  };

  onMove = event => {
    this.setState({
      lastMouseEvent: event,
    });
  };

  getFeatures = event => {
    if (event.features.length > 0) {
      return event.features;
    }
    return [];
  };

  updateLayers = (prevLayers = []) => {
    const { layers } = this.props;
    if (prevLayers.length > 0) {
      prevLayers.forEach(layer => {
        if (!layers.includes(layer)) {
          this.off(layer);
        }
      });
    }

    if (layers.length > 0) {
      layers.forEach(layer => {
        if (!prevLayers.includes(layer)) {
          this.on(layer);
        }
      });
    }
  };

  on = layer => {
    const { map } = this.props;

    map.on('mouseover', layer, this.onMouseOver);
    map.on('mouseleave', layer, this.onMouseLeave);
  };

  off = layer => {
    const { map } = this.props;

    map.off('mouseover', layer, this.onMouseOver);
    map.off('mouseleave', layer, this.onMouseLeave);
    map.off('move', this.onMove);
  };

  render() {
    const { children } = this.props;

    const { isHovering, lastMouseEvent, features } = this.state;

    if (!isHovering || !children) {
      return null;
    }

    return children(features, lastMouseEvent);
  }
}

export default function Hover(props) {
  return <MapConsumer>{map => <MapHover {...props} map={map} />}</MapConsumer>;
}
