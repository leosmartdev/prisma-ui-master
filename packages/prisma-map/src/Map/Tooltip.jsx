/*
 * We disable this eslint rule becaue its not picking up we are using the props in the new React
 * lifecycle method getDerivedStateFromProps.
 */
/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import { center } from '@turf/turf';
import { getCoord } from '@turf/invariant';

import { Consumer as MapConsumer } from './MapContext';

class Tooltip extends React.Component {
  static propTypes = {
    point: PropTypes.shape({
      x: PropTypes.number.isRequired,
      y: PropTypes.number.isRequired,
    }),
    coordinates: PropTypes.shape({
      lat: PropTypes.number,
      lng: PropTypes.number,
      latitude: PropTypes.number,
      longitude: PropTypes.number,
    }),
    feature: PropTypes.shape({
      type: PropTypes.oneOf(['Feature']),
      geometry: PropTypes.shape({
        type: PropTypes.string.isRequired,
        coordinates: PropTypes.arrayOf(PropTypes.number).isRequired,
      }),
      properties: PropTypes.object,
    }),
    /**
     * placement location of the tooltip in relation to the mouse.
     * Valid values are:
     *
     * ----------------------------------------------
     * | top-left        top-center       top-right |
     * |                                            |
     * | middle-left        mouse      middle-right |
     * |                                            |
     * | bottom-left   bottom-center   bottom-right |
     * ---------------------------------------------|
     * /
    placement: PropTypes.oneOf([
      'top-left',
      'top-center',
      'top-right',
      'middle-left',
      'middle-center',
      'middle-right',
      'bottom-left',
      'bottom-center',
      'bottom-right',
    ]),
    */
    children: PropTypes.oneOfType([
      PropTypes.element,
      PropTypes.func,
      PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.element, PropTypes.func])),
    ]),
    /** @private withStyles */
    classes: PropTypes.object.isRequired,
  };

  static defaultProps = {
    feature: null,
    point: null,
    coordinates: null,
    // placement: 'bottom-right',
    children: null,
  };

  static getDerivedStateFromProps(props, prevState) {
    if (
      props.feature !== prevState.feature ||
      props.coordinates !== prevState.coordinates ||
      props.point !== prevState.point
    ) {
      return {
        feature: props.feature,
        coordinates: props.coordinates,
        point: Tooltip.getPointFromProps(props),
      };
    }

    return null;
  }

  static getPointFromProps(props) {
    if (props.point !== null) {
      return props.point;
    }

    let point = null;
    let coord = null;
    const { coordinates, feature } = props;

    if (coordinates) {
      coord = [coordinates.lng || coordinates.longitude, coordinates.lat || coordinates.latitude];
    } else if (feature) {
      coord = center(props.feature);
    }

    if (coord && props.map) {
      const lngLat = getCoord(coord);
      point = props.map.project(lngLat);
    }

    return point;
  }

  state = {
    point: null,
  };

  /* TODO: this wasn't working, so we will come back to handling placement
  calculatePlacementOffsets = (placement) => {
    let xOffset = '0px';
    let yOffset = '0px';
    let transformOrigin = '';

    switch(placement) {
      case 'top-left': {
      }
      case 'top-middle': {
      }
      case 'top-right': {
      }
      case 'middle-left': {
      }
      case 'middle-right': {
      }
      case 'bottom-left': {
      }
      case 'bottom-middle': {
      }
      case 'bottom-right': {
        xOffset = '8px';
        yOffset = '16px';
        break;
      }
    }


    return { xOffset, yOffset, transformOrigin };
  }
  */

  render() {
    const {
      children,
      classes,
      // placement,
    } = this.props;

    const { point } = this.state;

    if (!point) {
      return null;
    }

    // TODO: need to get transform origin to work so we can accurately calculate
    // placement, so for now, lets just offset so the mouse doesnt' cover and leave it.
    // const { xOffset, yOffset, transformOrigin } = this.calculatePlacementOffsets(placement);
    const xOffset = '15px';
    const yOffset = '15px';
    const translateX = `calc(${point.x}px + ${xOffset})`;
    const translateY = `calc(${point.y}px + ${yOffset})`;

    return (
      <div
        className={classes.root}
        style={{
          transform: `translate(${translateX}, ${translateY})`,
        }}
      >
        {children}
      </div>
    );
  }
}

export const MapTooltip = withStyles({
  root: {
    position: 'absolute',
  },
})(Tooltip);

export default props => <MapConsumer>{map => <MapTooltip {...props} map={map} />}</MapConsumer>;
