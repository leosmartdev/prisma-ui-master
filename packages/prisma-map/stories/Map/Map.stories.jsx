import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, number } from '@storybook/addon-knobs';
import BasicUsage from './BasicUsage.md';

import Map from '../../src/Map';

const mapStories = storiesOf('Components/Map', module);

mapStories.add('Basic Component', () => <Map />, { notes: BasicUsage, info: { inline: true } });

mapStories.add('Change Map Size', () => <Map width={600} height={200} />);

mapStories.add('"Full Screen Map"', () => <Map fullScreen />);

mapStories.add(
  'Viewport Properties',
  () => (
    <Map
      latitude={number('latitude', 37.799, { min: -90, max: 90 })}
      longitude={number('longitude', -122.444, { min: -180, max: 180 })}
      zoom={number('zoom', 12, { min: 0, max: 24 })}
      pitch={number('pitch', 60, { min: 0, max: 60 })}
      bearing={number('bearing', 45, { min: 0, max: 359 })}
      width={number('width', 500)}
      height={number('height', 500)}
      fullScreen={boolean('fullScreen', false)}
    />
  ),
  { info: { inline: true } },
);
