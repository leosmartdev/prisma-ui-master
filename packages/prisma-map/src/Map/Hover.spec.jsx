import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import mapboxgl from 'mapbox-gl';
import { setupMapboxGlMocks } from '../../test/mapboxMock';
import Hover, { MapHover } from './Hover';
import Map from './Map';
import { Provider as MapProvider } from './MapContext';

// mock the mapbox library
jest.mock('mapbox-gl');

describe('Hover default export', () => {
  beforeEach(() => {
    setupMapboxGlMocks(mapboxgl);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('Renders', () => {
    const map = {
      on: jest.fn(),
      off: jest.fn(),
    };

    mount(<MapProvider value={map}><Hover /></MapProvider>);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Map mapId="hover"><Hover /></Map>);
    expect(tree).toMatchSnapshot();
  });
});

describe('Hover.MapHover', () => {
  const map = {
    on: jest.fn(),
    off: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Renders', () => {
    shallow(<MapHover map={map} />);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<MapHover map={map} />);
    expect(tree).toMatchSnapshot();
  });

  it('map not passed in', () => {
    const wrapper = shallow(<MapHover map={null} layers={['foo']} />);

    wrapper.setProps({ layers: ['bar'] });
    // If map == null is not properly handled, the test will fail with an exception at this point.
  });

  it('subscribes to events when map is no longer null', () => {
    const wrapper = shallow(<MapHover map={null} layers={['foo']} />);

    expect(map.on).not.toBeCalled();
    expect(map.off).not.toBeCalled();

    wrapper.setProps({ map });

    expect(map.on).toBeCalledWith('mouseover', 'foo', expect.any(Function));
    expect(map.on).toBeCalledWith('mouseleave', 'foo', expect.any(Function));
    expect(map.off).not.toBeCalled();
  });

  describe('layers property', () => {
    it('subscribes to events when layer property has a layer', () => {
      const layers = ['foo'];
      shallow(<MapHover map={map} layers={layers} />);

      expect(map.on).toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.on).toBeCalledWith('mouseleave', 'foo', expect.any(Function));
      expect(map.off).not.toBeCalled();
    });

    it('subscribes to layers events when a new layer is added', () => {
      const layers = ['foo'];
      const wrapper = shallow(<MapHover map={map} layers={layers} />);

      wrapper.setProps({ layers: [...layers, 'bar'] });

      expect(map.on).toBeCalledWith('mouseover', 'bar', expect.any(Function));
      expect(map.on).toBeCalledWith('mouseleave', 'bar', expect.any(Function));
    });

    it('unsubscribes to layers events when a layer is removed', () => {
      const layers = ['foo'];
      const wrapper = shallow(<MapHover map={map} layers={layers} />);

      jest.resetAllMocks();

      wrapper.setProps({ layers: [] });

      expect(map.off).toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.off).toBeCalledWith('mouseleave', 'foo', expect.any(Function));
      expect(map.off).toBeCalledWith('move', expect.any(Function));
      expect(map.on).not.toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.on).not.toBeCalledWith('mouseleave', 'foo', expect.any(Function));
    });

    it('only unsubscribes to layers that were removed', () => {
      const layers = ['foo', 'bar'];
      const wrapper = shallow(<MapHover map={map} layers={layers} />);

      jest.resetAllMocks();

      wrapper.setProps({ layers: ['bar'] });

      expect(map.off).toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.off).toBeCalledWith('mouseleave', 'foo', expect.any(Function));
      expect(map.off).not.toBeCalledWith('mouseover', 'bar', expect.any(Function));
      expect(map.off).not.toBeCalledWith('mouseleave', 'bar', expect.any(Function));
    });

    it('only subscribes to layers that were added', () => {
      const layers = ['foo'];
      const wrapper = shallow(<MapHover map={map} layers={layers} />);

      expect(map.on).toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.on).toBeCalledWith('mouseleave', 'foo', expect.any(Function));

      jest.resetAllMocks();

      wrapper.setProps({ layers: ['foo', 'bar'] });

      expect(map.on).not.toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.on).not.toBeCalledWith('mouseleave', 'foo', expect.any(Function));
      expect(map.on).toBeCalledWith('mouseover', 'bar', expect.any(Function));
      expect(map.on).toBeCalledWith('mouseleave', 'bar', expect.any(Function));
    });

    it('doesnt change listeners if the same layers are passed in', () => {
      const layers = ['foo', 'bar'];
      const wrapper = shallow(<MapHover map={map} layers={layers} />);

      expect(map.off).not.toBeCalled();
      expect(map.on).toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.on).toBeCalledWith('mouseleave', 'foo', expect.any(Function));
      expect(map.on).toBeCalledWith('mouseover', 'bar', expect.any(Function));
      expect(map.on).toBeCalledWith('mouseleave', 'bar', expect.any(Function));

      jest.resetAllMocks();

      wrapper.setProps({ layers: ['foo', 'bar'] });

      expect(map.off).not.toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.off).not.toBeCalledWith('mouseleave', 'foo', expect.any(Function));
      expect(map.off).not.toBeCalledWith('mouseover', 'bar', expect.any(Function));
      expect(map.off).not.toBeCalledWith('mouseleave', 'bar', expect.any(Function));
      expect(map.on).not.toBeCalledWith('mouseover', 'foo', expect.any(Function));
      expect(map.on).not.toBeCalledWith('mouseleave', 'foo', expect.any(Function));
      expect(map.on).not.toBeCalledWith('mouseover', 'bar', expect.any(Function));
      expect(map.on).not.toBeCalledWith('mouseleave', 'bar', expect.any(Function));
    });
  });

  describe('onMouseOver', () => {
    it('subscribes to move to track new changes', () => {
      const event = {
        features: [],
      };
      const wrapper = shallow(<MapHover map={map} layers={['foo']} />);

      wrapper.instance().onMouseOver(event);

      expect(map.on).toHaveBeenCalledWith('move', expect.any(Function));
    });

    it('saves features and event to the state', () => {
      const event = {
        features: [{ id: 'foo' }, { id: 'bar' }],
      };
      const wrapper = shallow(<MapHover map={map} layers={['foo']} />);

      wrapper.instance().onMouseOver(event);

      expect(wrapper.state('features')).toEqual([{ id: 'foo' }, { id: 'bar' }]);
      expect(wrapper.state('lastMouseEvent')).toEqual(event);
    });

    it('sets is hovering to true', () => {
      const event = { features: [] };
      const wrapper = shallow(<MapHover map={map} layers={['foo']} />);

      expect(wrapper.state('isHovering')).toBe(false);

      wrapper.instance().onMouseOver(event);

      expect(wrapper.state('isHovering')).toBe(true);
    });
  });

  describe('onMouseLeave', () => {
    it('unsubscribes to move event', () => {
      const wrapper = shallow(<MapHover map={map} layers={['foo']} />);

      wrapper.instance().onMouseLeave();

      expect(map.off).toHaveBeenCalledWith('move', expect.any(Function));
    });

    it('clears features and event on state', () => {
      const event = { features: [] };
      const wrapper = shallow(<MapHover map={map} layers={['foo']} />);

      wrapper.instance().onMouseOver(event);
      expect(wrapper.state('features')).toEqual(event.features);
      expect(wrapper.state('lastMouseEvent')).toEqual(event);

      wrapper.instance().onMouseLeave();

      expect(wrapper.state('features')).toEqual(null);
      expect(wrapper.state('lastMouseEvent')).toEqual(null);
    });

    it('sets is hovering to false', () => {
      const event = { features: [] };
      const wrapper = shallow(<MapHover map={map} layers={['foo']} />);

      wrapper.instance().onMouseOver(event);

      expect(wrapper.state('isHovering')).toBe(true);

      wrapper.instance().onMouseLeave();

      expect(wrapper.state('isHovering')).toBe(false);
    });
  });

  describe('onMove', () => {
    it('updates lastMouseEvent', () => {
      const event = {
        features: [],
      };
      const wrapper = shallow(<MapHover map={map} layers={['foo']} />);

      wrapper.setState({ features: [{ id: 'foo' }] });
      expect(wrapper.state('lastMouseEvent')).toEqual(null);
      expect(wrapper.state('features')).toEqual([{ id: 'foo' }]);
      wrapper.instance().onMove(event);
      expect(wrapper.state('lastMouseEvent')).toEqual(event);
      expect(wrapper.state('features')).toEqual([{ id: 'foo' }]);
    });
  });

  describe('children', () => {
    it('doesnt render child when mouse is not hovering', () => {
      const child = jest.fn();
      shallow((
        <MapHover map={map} layers={['foo']}>
          {child}
        </MapHover>
      ));

      expect(child).not.toHaveBeenCalled();
    });

    it('renders child when mouse is hovering', () => {
      const event = {
        features: [{ id: 'foo' }, { id: 'bar' }],
      };
      const child = jest.fn();
      const wrapper = shallow((
        <MapHover map={map} layers={['foo']}>
          {child}
        </MapHover>
      ));

      wrapper.setState({
        isHovering: true,
        features: event.features,
        lastMouseEvent: event,
      });

      expect(child).toHaveBeenCalledWith(event.features, event);
    });
  });
});
