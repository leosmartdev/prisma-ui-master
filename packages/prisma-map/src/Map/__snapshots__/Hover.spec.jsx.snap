// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hover default export matches snapshot 1`] = `
<div
  style={
    Object {
      "height": 500,
      "position": "relative",
      "width": 800,
    }
  }
>
  <div
    className="prisma-map-div"
    id="hover"
    style={
      Object {
        "height": 500,
        "width": 800,
      }
    }
  />
  <div
    id="map-controls-overlay"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "overflow": "hidden",
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</div>
`;

exports[`Hover.MapHover matches snapshot 1`] = `null`;
